spec/models/schedule_rule_spec.rb

```diff
@@ -12,6 +15,52 @@
     @rule.should be_valid
   end
 
+  describe ".unavailable_for_date" do
+    context "for an instrument available only from 9 AM to 5 PM" do
+      before(:each) do
+        instrument.schedule_rules.create(attributes_for(:schedule_rule))
+      end
+
+      let(:reservations) { ScheduleRule.unavailable_for_date(instrument, day) }
+
+      shared_examples_for "it generates reservations to cover unavailability" do
+        it "returns two dummy reservations" do
+          expect(reservations.size).to eq(2)
+          reservations.each do |reservation|
+            expect(reservation).to be_kind_of(Reservation)
+            expect(reservation).not_to be_persisted
```

Comment:
Do we have the rspec feature: `expect(reservations).to all( be_kind_of(Reservation )`)?